gdal/doc/source/development/rfc/rfc77_drop_python2_support.rst

```diff
@@ -0,0 +1,214 @@
+.. _rfc-77:
+
+================================================================================
+RFC 77: Drop Python 2 support in favour of Python 3.6|3.7
+================================================================================
+
+============== ============================
+Author:        Idan Miara
+Contact:       idan@miara.com
+Started:       2020-Nov-3
+Last updated:  2020-Nov-3
+Status:        Preliminary, Target GDAL 3.3
+============== ============================
+
+Summary
+-------
+
+This RFC drops Python 2 support.
+This RFP sets Python 3.6 (or 3.7) as the new minimum supported Python version.
+
+Motivation
+----------
+
+Currently GDAL Python bindings support Python 2.7 and Python 3.x (x=?).
+Python 2 is at End Of Life, and is no longer supported since the January 2020.
+Keeping Python 2.7 support virtually serves no purpose as virtually all major OS
+and relevant programs already dropped Python 2 support.
+On the other hand, keeping Python 2 support puts unnecessary maintenance burden on the GDAL maintainers,
+As maintainers need to make sure their new code is backwards compatible with Python 2.
+Moreover, many important features that added in Python 3 cannot be used in GDAL to maintain backwards compatibility with Python 2.
+
+Incomplete features list of Python 3.6, several of which are immediately useful to simplify code or improve testing:
```

Comment:
this is only features of 3.6 cf 3.5... not 3.x over 2.7 - of which there are many more.

Reply:
Right, Do you want to list any specific ones that worth noting ?